src/lib.rs

```diff
@@ -200,6 +203,7 @@ pub struct Node {
 	scorer: Arc<Mutex<Scorer>>,
 	peer_store: Arc<PeerStore<Arc<Logger>>>,
 	payment_store: Arc<PaymentStore>,
+	is_running: Arc<RwLock<bool>>,
```

Comment:
No AtomicBool this time?

Reply:
I see. Otherwise you'd allow things like concurrent start calls, or the node being used before start is finished.